examples/save_data_example.py

```diff
@@ -86,24 +72,44 @@ def run_scans(nScans, power_idx, field_idx, DNP_data=None):
         p.set_power(this_dB)
         for k in range(10):
             time.sleep(short_delay)
-            if p.get_power_setting() >= this_dB: break
+            if p.get_power_setting() >= this_dB: 
+                break
         time.sleep(long_delay)
-        meter_powers[j] = p.get_power_setting()
-        if True:
-            # the following seems unrealistic for a field sweep -- what's
-            # up with that?
-            for B0_index,desired_B0 in enumerate(r_[3501:3530:1]):
-                #carrierFreq_MHz = rand()
-                carrierFreqs_MHz[B0_index] = rand()
-                fields_Set[B0_index] = rand()
-                time.sleep(short_delay)
-                if True:
-                    DNP_data = run_scans(1,j,B0_index,DNP_data)
-                    time.sleep(long_delay)
-    log_array, log_dict = p.stop_log()# where error occurred originally!
-print("EXITING...")    
-#}}}
+        power_settings_dBm[j] = p.get_power_setting()
+        DNP_ini_time = time.time()
+        if j == 0: 
+            retval = p.dip_lock(
+                config_dict['uw_dip_center_GHz'] - config_dict['uw_dip_width_GHz'] / 2,
+                config_dict['uw_dip_center_GHz'] + config_dict['uw_dip_width_GHz'] / 2,
+            ) #needed to set powers above 10 dBm - in future we plan on debugging so this is not needed
+            DNP_data = run_scans(
+                    indirect_idx=j,
+                    indirect_len=len(powers),
+                    indirect_fields=("start_times", "stop_times"),
+                    ret_data=None,
+                    )
+        else:
+            run_scans(
+                    indirect_idx=j,
+                    indirect_len=len(powers),
+                    indirect_fields=("start_times", "stop_times"),
+                    ret_data=DNP_data,
+                    )
+        DNP_done = time.time()
```

Comment:
normally I would set DNP_data to None outside the loop (as it's done in the combined_ODNP.py script) but it was then giving me the error that DNP_data had Nonetype and I couldn't for the life of me figure out why. So this works

Reply:
confusing comment